spec.html

```diff
@@ -7870,11 +7893,10 @@ <h1>ArrayCreate ( _length_ [ , _proto_ ] )</h1>
           1. If _length_ is *-0*, set _length_ to *+0*.
           1. If _length_ &gt; 2<sup>32</sup> - 1, throw a *RangeError* exception.
           1. If _proto_ is not present, set _proto_ to the intrinsic object %ArrayPrototype%.
-          1. Let _A_ be a newly created Array exotic object.
-          1. Set _A_'s essential internal methods except for [[DefineOwnProperty]] to the default ordinary object definitions specified in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>.
-          1. Set _A_.[[DefineOwnProperty]] as specified in <emu-xref href="#sec-array-exotic-objects-defineownproperty-p-desc"></emu-xref>.
+          1. Let _A_ be ! AllocateBasicObject(&laquo; [[Prototype]], [[Extensible]] &raquo;).
           1. Set _A_.[[Prototype]] to _proto_.
-          1. Set _A_.[[Extensible]] to *true*.
+          1. Set _A_.[[DefineOwnProperty]] as specified in <emu-xref href="#sec-array-exotic-objects-defineownproperty-p-desc"></emu-xref>.
+          1. Assert: _A_ is an Array exotic object.
```

Comment:
This is the only case where you added such an assertion. Did you mean to add it for the other exotic Creates?